embedded-wallets/sdk/android/migration-guides/android-v9-to-v10.mdx

```diff
@@ -0,0 +1,414 @@
+        idToken = "your_jwt_token"
+    )
+)
+
+// SFA with aggregate verifier
+val loginCompletableFuture = web3Auth.connectTo(
+    LoginParams(
+        AuthConnection.CUSTOM,
+        authConnectionId = "aggregate_verifier_id",
+        groupedAuthConnectionId = "sub_verifier_id",
+        idToken = "your_jwt_token"
+    )
+)
+```
+
+### 2. Enhanced Parameter Handling
+
+v10.x provides better parameter organization:
+
+```kotlin
+val loginCompletableFuture = web3Auth.connectTo(
+    LoginParams(
+        authConnection = AuthConnection.EMAIL_PASSWORDLESS,
+        loginHint = "user@example.com", // Direct parameter
+        dappUrl = "https://your-dapp.com", // For analytics
+        appState = "custom_state" // App state tracking
+    )
+)
+```
+
+### 3. Automatic Chain Configuration
+
+Chain configuration is now managed automatically from your Web3Auth Dashboard:
+
+1. Configure chains in your [Web3Auth Dashboard](https://dashboard.web3auth.io)
+2. The SDK automatically uses these configurations
+3. No need to manually pass `ChainConfig` to wallet services
+
+## Migration Steps
+
+### Step 1: Update Dependencies
+
+```groovy
+// In your app-level build.gradle
+dependencies {
+    implementation 'com.github.web3auth:web3auth-android-sdk:10.0.0'
+}
+```
+
+### Step 2: Update Imports
+
+Replace old imports with new ones as shown in the [Import Changes](#2-import-changes) section.
+
+### Step 3: Update Initialization
+
+1. Replace `network` with `web3AuthNetwork`
+2. Replace `redirectUrl` Uri with String
+3. Update `loginConfig` to `authConnectionConfig`
+4. Replace `buildEnv` with `authBuildEnv`
+
+### Step 4: Update Authentication Calls
+
+1. Replace `login()` with `connectTo()`
+2. Replace `Provider` with `AuthConnection`
+3. Update parameter names in `LoginParams`
+
+### Step 5: Update Private Key Calls
+
+1. Replace `getPrivKey()` with `getPrivateKey()`
+2. Replace `getEd25519PrivKey()` with `getEd25519PrivateKey()`
+
+### Step 6: Update Wallet Services
+
+1. Replace `launchWalletServices()` with `showWalletUI()`
+2. Remove `ChainConfig` parameters from both `showWalletUI()` and `request()`
+
+### Step 7: Configure Dashboard
+
+Ensure your Web3Auth Dashboard has the necessary chain configurations since they're now automatically managed.
+
+## Troubleshooting
+
+### Common Issues
+
+1. **Chain Configuration Missing**: Ensure your Web3Auth Dashboard has the required chains configured
+2. **Import Errors**: Make sure to import `Web3AuthNetwork` from the correct package
+3. **Parameter Mismatch**: Double-check that you're using the new parameter names
+
+### Migration Checklist
+
+- [ ] Dependencies updated to v10.x
+- [ ] Imports updated with new package references
+- [ ] `Web3AuthOptions` configuration updated
+- [ ] `login()` calls replaced with `connectTo()`
+- [ ] Private key method names updated
+- [ ] Wallet services methods updated
+- [ ] Dashboard chain configuration verified
+- [ ] Testing completed for all authentication flows
```

Comment:
Btw, this doesn't display as a checklist in the rendered markdown: https://metamask-docs-mq9lyhz8p-consensys-ddffed67.vercel.app/embedded-wallets/sdk/android/migration-guides/android-v9-to-v10/#migration-checklist